### PR TITLE
Add Visual Novel Example to Example branch

### PR DESCRIPTION
**Pull request by Crystalwarrior**
Add a visual novel example, including a test timeline showing off blips, background transitions, etc.

### CODE REVIEW

**Comment by Crystalwarrior** [comment]
how tf do I convert to draft

File: examples/VisualNovel/Main.gd [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		elif event_manager.timeline in seen_events and \
+			 current_event in seen_events[event_manager.timeline]:
+				dialog_manager.display_all_text()
+
+	if Input.is_action_just_pressed("ui_down"):
+		bg_index = (bg_index + 1) % bgs.size()
+		set_background(bgs[bg_index], 0, 0)
+
+
+func set_background(bg_path: String, smooth: float = 0.5, speed: float = 1.0):
```
Comment (AnidemDex) [inline examples/VisualNovel/Main.gd:43]: an event for this function could be a good idea?

**Review by AnidemDex — COMMENTED** [review]
I think it needs more comments to know what you were doing _(I can understand what is happening here, but since we're on the examples branch, we need to do it with ELI5 as possible)_

File: examples/VisualNovel/Main.gd (hunk 1 above)
Comment (Crystalwarrior) [reply examples/VisualNovel/Main.gd:43]: Very true! I was considering creating events from the example itself, but it would defo be better to just make it part of Textalog.

File: examples/VisualNovel/DialogPiece.tscn [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+margin_bottom = 14.0
+text = "Character Name"
+
+[node name="Dialog" type="RichTextLabel" parent="HBoxContainer/VBoxContainer"]
+margin_top = 18.0
+margin_right = 942.0
+margin_bottom = 64.0
+size_flags_horizontal = 3
+size_flags_vertical = 3
+text = "Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text Text text text "
```
[the remark this answers is not in the document]
Comment (Crystalwarrior) [reply examples/VisualNovel/DialogPiece.tscn:39]: Placeholder text for the dialog piece, can be changed to just "" tbh

**Comment by Crystalwarrior** [comment]
> I think it needs more comments to know what you were doing _(I can understand what is happening here, but since we're on the examples branch, we need to do it with ELI5 as possible)_

I agree. I will excessively comment it once I have the code as modular and nice to read as possible

File: examples/VisualNovel/Main.gd (hunk 1 above)
Comment (AnidemDex) [reply examples/VisualNovel/Main.gd:43]: You can make it an event of textalog, adding the event to events folder and its path reference to the plugin script.

**Comment by AnidemDex** [comment]
Gonna merge this branch after you change the "text text text" placeholder with a proper one (like "placeholder text" or just ""). The other two revisions can be made in a different branch

File: examples/VisualNovel/Main.gd [hunk 3]
```diff
@@ -56,16 +63,26 @@ func set_background(bg_path: String, smooth: float = 0.5, speed: float = 1.0):
 
 func _on_EventManager_event_started(event: Event):
 	current_event = event
+	if event is EventDialogText:
+		current_character = event.character
+		if current_character:
+			var node = dialog_node.portrait_manager.portraits[current_character]
```
Comment (AnidemDex) [inline examples/VisualNovel/Main.gd:69]: Woops, I forgot to open the issue to add in PortraitManager a method to get the node related to the character

File: examples/VisualNovel/portrait_manager_ex.gd [hunk 4]
```diff
@@ -0,0 +1,127 @@
+extends PortraitManager
```
Comment (AnidemDex) [inline examples/VisualNovel/portrait_manager_ex.gd:1]: Why this extention is a duplicate of the original script?

**Review by AnidemDex — CHANGES_REQUESTED** [review]
A few points:
- Restore `.gdignore` file
- Are `psd`, `xcf`, `lnk` files necessary?

File: examples/VisualNovel/portrait_manager_ex.gd (hunk 4 above)
Comment (Crystalwarrior) [reply examples/VisualNovel/portrait_manager_ex.gd:1]: Wdym duplicate? The script is different because the original script does not support non-TextureRect portraits

**Comment by Crystalwarrior** [comment]
> A few points:
> 
>     * Restore `.gdignore` file
> 
>     * Are `psd`, `xcf`, `lnk` files necessary?

.psd, .xcf, etc. files are not really necessary, they could be distributed separately. Source files can be helpful for creators to understand how the sprites are composed tho.

File: examples/VisualNovel/Main.gd (hunk 3 above)
Comment (AnidemDex) [reply examples/VisualNovel/Main.gd:68]: https://github.com/AnidemDex/Godot-DialogPlugin/issues/73

File: examples/VisualNovel/Main.gd [hunk 5]
```diff
@@ -42,9 +42,8 @@ func _process(delta):
 		set_background(bgs[bg_index], 0, 0)
 
 
-func set_music(mus_path: String):
-	music_player.stream = load(mus_path)
-	music_player.play()
+func set_music(mus_path: String, speed: float = 1.0):
+	music_player.play_music(load(mus_path), speed)
```
Comment (AnidemDex) [inline examples/VisualNovel/Main.gd:46]: `set_` function to `play` it inmediatly?

Comment (Crystalwarrior) [reply examples/VisualNovel/Main.gd:46]: you're right, should be named "play_music"

**Comment by AnidemDex** [comment]
don't forget to `rebase main` and force push your branch when you're done to solve merge conflicts

File: addons/textalog/assets/themes/default_theme/plain_black.tres [hunk 6]
```diff
@@ -86,5 +86,10 @@ Button/styles/hover = SubResource( 3 )
 Button/styles/normal = SubResource( 4 )
 Button/styles/pressed = SubResource( 5 )
 DialogNode/styles/bubble = null
+<<<<<<< HEAD
```
Comment (AnidemDex) [inline addons/textalog/assets/themes/default_theme/plain_black.tres:89]: Diff commit header. This should be solved before adding more commits

Comment (Crystalwarrior) [reply addons/textalog/assets/themes/default_theme/plain_black.tres:89]: aw god damnit, thought I resolved that

**Review by AnidemDex — APPROVED** [review]
Seems good

**Comment by Crystalwarrior** [comment]
Worked on my game, gotta backport a lot of new discoveries I made during its development into this example later

**Comment by AnidemDex** [comment]
Please, mark it as done when you think is ready to be merged.
